resolve/resolve_test.go

```diff
@@ -0,0 +1,126 @@
+package resolve
+
+import (
+	"testing"
+
+	"github.com/bazelbuild/bazel-gazelle/rule"
+	"github.com/bazelbuild/bazel-gazelle/config"
+	"github.com/bazelbuild/bazel-gazelle/label"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestFindRuleWithOverride_ParentTraversal(t *testing.T) {
+	rootCfg := getConfig(t, "", []rule.Directive{
+		{Key: "resolve", Value: "go go github.com/root/repo @com_example//root:replacement"},
+		{Key: "resolve_regexp", Value: "go ^github.com/root/(.*)$ @com_example//regexp:replacement"},
```

Comment:
If need be, we could optimize these special types of (probably common) "prefix" regexp resolve directives further.

Reply:
True, this might be a bit complex logic. I think I'd prefer to wait and do this in a later PR.

Reply:
Yes, fully agree with that